api/go.mod

```diff
@@ -2,30 +2,18 @@ module github.com/metal3-io/cluster-api-provider-metal3/api
 
 go 1.16
 
-replace k8s.io/client-go => k8s.io/client-go v0.17.9
-
-replace k8s.io/apimachinery => k8s.io/apimachinery v0.17.9
-
-replace k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.17.9
-
-replace k8s.io/api => k8s.io/api v0.17.9
-
-replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
-
-replace github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.0
-
-replace sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.5.14
-
 require (
 	github.com/google/gofuzz v1.2.0
 	github.com/metal3-io/ip-address-manager/api v0.0.0-20210609163946-48b0ce9a1ac0
 	github.com/onsi/gomega v1.13.0
 	github.com/pkg/errors v0.9.1
 	golang.org/x/net v0.0.0-20210525063256-abc453219eb5
-	k8s.io/api v0.20.2
-	k8s.io/apimachinery v0.20.2
+	k8s.io/api v0.21.1
+	k8s.io/apimachinery v0.21.1
 	k8s.io/client-go v1.5.1
 	k8s.io/utils v0.0.0-20210527160623-6fdb442a123b
-	sigs.k8s.io/cluster-api v0.3.17
-	sigs.k8s.io/controller-runtime v0.8.3
+	sigs.k8s.io/cluster-api v0.4.0-beta.0
```

Comment:
I think this is overlapping with https://github.com/metal3-io/cluster-api-provider-metal3/pull/167/files#diff-ba0784a950902285760f839a4ad720a381b2981e8a67147f5f43f7508e6343ed. What's the plan with CAPI v1a4 @furkatgofurov7 @asalkeld ? Shall we wait until https://github.com/metal3-io/cluster-api-provider-metal3/pull/167 lands and rebase this work ?

Reply:
Most probably 167 first since that is being tested currently from [m3-dev-env](https://github.com/metal3-io/metal3-dev-env/pull/693) PR.